core/eolearn/core/eodata_io.py

```diff
@@ -0,0 +1,260 @@
+"""
+A module implementing utilities for working with saving and loading EOPatch data
+
+Credits:
+Copyright (c) 2017-2020 Matej Aleksandrov, Matej Batič, Grega Milčinski, Matic Lubej, Devis Peresutti (Sinergise)
+Copyright (c) 2017-2020 Nejc Vesel, Jovan Višnjić, Anže Zupanc (Sinergise)
+
+This source code is licensed under the MIT license found in the LICENSE
+file in the root directory of this source tree.
+"""
+import pickle
+import gzip
+import concurrent.futures
+
+import fs
+from fs.tempfs import TempFS
+import numpy as np
+from sentinelhub.os_utils import sys_is_windows
+
+from .constants import FeatureType, FileFormat, OverwritePermission
+from .utilities import FeatureParser
+
+
+def save_eopatch(eopatch, filesystem, patch_location, features=..., overwrite_permission=OverwritePermission.ADD_ONLY,
+                 compress_level=0):
+    """ A utility function used by EOPatch.save method
+    """
+    patch_exists = filesystem.exists(patch_location)
+
+    if overwrite_permission is OverwritePermission.OVERWRITE_PATCH and patch_exists:
+        filesystem.removetree(patch_location)
+        patch_exists = False
+
+    if not patch_exists:
+        filesystem.makedir(patch_location, recreate=True)
```

Comment:
This was because of a special case when `patch_location='/'` and `overwrite_permission=OverwritePermission.OVERWRITE_PATCH`. But I see it wasn't a very nice solution and I changed how we handle that case. Now we just use `filesystem.makedir(patch_location)` which by default sets `recreate=False`.